oauthenticator/cilogon.py

```diff
@@ -241,144 +273,105 @@ def _validate_allowed_idps(self, proposal):
 
         return username
 
-    def user_info_to_username(self, user_info):
-        username_claims = self._get_final_username_claim_list(user_info)
-        username = self._get_username_from_claim_list(user_info, username_claims)
+    def _user_info_to_unprocessed_username(self, user_info):
+        """
+        Returns a username from
+        """
+        user_idp = user_info["idp"]
+        username_derivation = self.allowed_idps[user_idp]["username_derivation"]
+        username_claim = username_derivation["username_claim"]
 
+        username = user_info.get(username_claim)
         if not username:
-            user_info_keys = sorted(user_info.keys())
-            self.log.error(
-                f"No username claim in the list at {username_claims} was found in the response {user_info_keys}"
-            )
-            raise web.HTTPError(500, "Failed to get username from CILogon")
-
-        # Optionally strip idp domain or prefix the username
-        if self.allowed_idps:
-            selected_idp = user_info["idp"]
-            if selected_idp in self.allowed_idps.keys():
-                username_derivation = self.allowed_idps[selected_idp][
-                    "username_derivation"
-                ]
-                action = username_derivation.get("action")
-
-                if action == "strip_idp_domain":
-                    username = username.split("@", 1)[0]
-                elif action == "prefix":
-                    prefix = username_derivation["prefix"]
-                    username = f"{prefix}:{username}"
+            message = f"Configured username_claim {username_claim} for {user_idp} was not found in the response {user_info.keys()}"
+            self.log.error(message)
+            raise web.HTTPError(500, message)
 
         return username
 
-    async def check_allowed(self, username, auth_model):
+    def _get_processed_username(self, username, user_info):
         """
-        Returns True for authorized users, raises errors for users
-        denied authorization.
-
-        Overrides the `OAuthenticator.check_allowed` implementation to only allow users
-        logging in using a provider that is  part of `allowed_idps`.
-        Following this, the user must either be part of `allowed_users` or `allowed_domains`
-        to be authorized if either is configured, otherwise all users are
-        authorized.
+        This method optionally adjusts a username from user_info based on the
+        "action" specified under "username_derivation" for the associated idp.
         """
-        # A workaround for JupyterHub<=4.0.1, described in
-        # https://github.com/jupyterhub/oauthenticator/issues/621
-        if auth_model is None:
-            return True
+        user_idp = user_info["idp"]
+        username_derivation = self.allowed_idps[user_idp]["username_derivation"]
+
+        # Optionally execute action "strip_idp_domain" or "prefix"
+        action = username_derivation.get("action")
```

Comment:
I believe the presence of the action itself must also be optional if for example we want the username to just be email claim with no processing whatsoever.

Reply:
Ignore this ⬆️ , this is already happening <3